File: IT/1/Funzioni/Esercizi/Python/esercizio10.py
# file esercizio10.py

from __future__ import print_function, division

def divide(a, b):
	return a / b if b is not 0 else float('nan')

a = 10
b = 3
c = 0

print("{} / {} = {:.3f}".format(a, b, divide(a, b)))
print("{} / {} = {}".format(a, c, divide(a, c)))

File: IT/1/Funzioni/Esercizi/Java/Esercizio10.java
/* file Esercizio10.java */

import java.util.Locale;

public class Esercizio10 {
	static float divide(int a, int b) {
		return b != 0 ? (float)a / b : Float.NaN;
	}

	public static void main(String [] args) {
		int a = 10;
		int b = 3;
		int c = 0;

		System.out.println(a + " / " + b + " = " + String.format(Locale.ROOT, "%.3f", divide(a, b)));
		System.out.println(a + " / " + c + " = " + divide(a, c));
	}
}

